In [1]:
import pandas as pd
from pymongo import MongoClient
from collections import defaultdict

In [2]:

# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

In [29]:
df_events = pd.read_csv("../data/EVENTS.csv", sep=";")
df_events.head()

_id                speaker_name  \
0    1                  Sol García   
1    2  Marta Paredes & Sol García   
2    3             Antonino Sistac   
3    4              Jorge Montañez   
4    5  Francesc Santanach Delisau   

                                     speaker_company  \
0                                    Directora Samoo   
1  Prof. Universidad Complutense de Madrid & Dire...   
2   Director of Iberia e Iberoamérica en ReadSpeaker   
3             Director of Iberoamerica Sales OpenLMS   
4                         Arquitecto Empresarial UOC   

                                          desc_event  id_place        date  \
0        ACTO DE APERTURA E-LEARNING EXPERIENCE 2024         1  2024-05-23   
1  TENDENCIAS Y PROSPECTIVA EN E-LEARNING PARA UN...         1  2024-05-23   
2  DERRIBEMOS BARRERAS EN LOS EXÁMENES Y CONTENID...         1  2024-05-23   
3         OPEN LMS SHOWCASE: LO NUEVO Y LO QUE VIENE         1  2024-05-23   
4  REDESCUBRIENDO LA EDUCACIÓN. UNA PROPUESTA DE ...         1  2024-05-23   

    hour id_users                                         interest  
0  10:30       []          ["E-learning","Educación","Innovación"]  
1  10:45       []          ["E-learning","Tendencias","Educación"]  
2  11:15       []             ["Educación","STEM","Accesibilidad"]  
3  11:45       []                ["LMS","Innovación","Tecnología"]  
4  12:15       []  ["Educación","Inteligencia Artificial","Datos"]

In [11]:
df_meetings = pd.read_csv("../data/MEETINGS.csv")
df_meetings.head()

_id  id_collab  id_user_collab  id_user        date   hour  feedback  table
0    1          1             283      148  2024-07-04  09:00         1     10
1    2          1             283      183  2024-07-04  09:45         4     23
2    3          1             266      211  2024-07-04  10:30         3      2
3    4          1             294      108  2024-07-04  11:15         3     30
4    5          1             287      131  2024-07-04  12:00         5      7

In [15]:
df_suppliers = pd.read_csv("../data/SUPPLIERS.csv", sep=";")
df_suppliers.head()


_id          CIF company_name  password address_contact         country  \
0    1  ES12345678X         Zoom  suppass1       Calle A 1          España   
1    2  ES23456789X        MLOps  suppass2       Calle B 2          España   
2    3  ES34567890X     Coursera  suppass3       Calle C 3  Estados Unidos   
3    4  ES45678901X        Udemy  suppass4       Calle D 4  Estados Unidos   
4    5  ES56789012X          edX  suppass5       Calle E 5     Reino Unido   

                  email  phone_prefx  phone_number type_collab  \
0      zoom@example.com           34     600000001   Platinium   
1     mlops@example.com           34     600000002        Gold   
2  coursera@example.com            1     600000003      Silver   
3     udemy@example.com            1     600000004   Platinium   
4       edx@example.com           44     600000005        Gold   

                     interests    id_users  tokens  
0   ["Tecnología","Educación"]  [266, 267]     NaN  
1   ["Innovación","Educación"]  [268, 269]     NaN  
2  ["E-learning","Tecnología"]       [270]     NaN  
3   ["E-learning","Educación"]  [271, 272]     NaN  
4   ["Educación","Innovación"]  [273, 274]     NaN

In [16]:
df_places = pd.read_csv("../data/PLACES.csv")
df_places

_id  id_place                       place_name  capacity
0    1         1  Sala Principal - La Font Blanca       150
1    2         2     Sala Workshops - La Alcazaba        80

In [17]:
df_users = pd.read_csv("../data/USERS.csv")
df_users.head()

_id    name  surname              email password  tokens   createdAt  \
0    1  Carlos    Gómez  user1@example.com    pass1     NaN  2022-01-01   
1    2    Jose     Díaz  user2@example.com    pass2     NaN  2022-01-01   
2    3   Maria    Pérez  user3@example.com    pass3     NaN  2022-01-01   
3    4    Luis    Gómez  user4@example.com    pass4     NaN  2022-01-01   
4    5    Juan  Sánchez  user5@example.com    pass5     NaN  2022-01-01   

    updatedAt  __v confirmed  ... phone_number      address  zip_code  \
0  2022-01-01    0       Yes  ...    615786083  Calle G 132     34144   
1  2022-01-01    0       Yes  ...    629613106    Calle B 4     96979   
2  2022-01-01    0       Yes  ...    602716833  Calle J 195     89142   
3  2022-01-01    0       Yes  ...    642871128   Calle A 19     20117   
4  2022-01-01    0       Yes  ...    673132819  Calle B 136     75320   

        city country  user_type                   url_linkedin  \
0   Valencia  España  Assistant  https://linkedin.com/in/user1   
1   Valencia  España  Assistant  https://linkedin.com/in/user2   
2   Valencia  España  Assistant  https://linkedin.com/in/user3   
3  Barcelona  España  Assistant  https://linkedin.com/in/user4   
4   Valencia  España  Assistant  https://linkedin.com/in/user5   

                          allergies  \
0            ['Sésamo', 'Moluscos']   
1  ['Lácteos', 'Pescado', 'Azucar']   
2                      ['Moluscos']   
3                                []   
4                                []   

                                           interests food_preferences  
0                                     ['Softskills']           vegano  
1                                     ['Tecnología']      vegetariano  
2                                       ['Negocios']      vegetariano  
3                  ['Negocios', 'Marketing Digital']           vegano  
4  ['Tecnología', 'Formación', 'Educación', 'Gest...      vegetariano  

[5 rows x 23 columns]

In [21]:
df_users.head().T

0  \
_id                                           1   
name                                     Carlos   
surname                                   Gómez   
email                         user1@example.com   
password                                  pass1   
tokens                                      NaN   
createdAt                            2022-01-01   
updatedAt                            2022-01-01   
__v                                           0   
confirmed                                   Yes   
company                  Emprendedor E-learning   
position                Facilitador de Talleres   
phone_prefx                                  34   
phone_number                          615786083   
address                             Calle G 132   
zip_code                                  34144   
city                                   Valencia   
country                                  España   
user_type                             Assistant   
url_linkedin      https://linkedin.com/in/user1   
allergies                ['Sésamo', 'Moluscos']   
interests                        ['Softskills']   
food_preferences                         vegano   

                                                 1  \
_id                                              2   
name                                          Jose   
surname                                       Díaz   
email                            user2@example.com   
password                                     pass2   
tokens                                         NaN   
createdAt                               2022-01-01   
updatedAt                               2022-01-01   
__v                                              0   
confirmed                                      Yes   
company                                  Colegio B   
position                       Gestor de Comunidad   
phone_prefx                                     34   
phone_number                             629613106   
address                                  Calle B 4   
zip_code                                     96979   
city                                      Valencia   
country                                     España   
user_type                                Assistant   
url_linkedin         https://linkedin.com/in/user2   
allergies         ['Lácteos', 'Pescado', 'Azucar']   
interests                           ['Tecnología']   
food_preferences                       vegetariano   

                                              2  \
_id                                           3   
name                                      Maria   
surname                                   Pérez   
email                         user3@example.com   
password                                  pass3   
tokens                                      NaN   
createdAt                            2022-01-01   
updatedAt                            2022-01-01   
__v                                           0   
confirmed                                   Yes   
company                               Colegio B   
position                 Responsable de Calidad   
phone_prefx                                  34   
phone_number                          602716833   
address                             Calle J 195   
zip_code                                  89142   
city                                   Valencia   
country                                  España   
user_type                             Assistant   
url_linkedin      https://linkedin.com/in/user3   
allergies                          ['Moluscos']   
interests                          ['Negocios']   
food_preferences                    vegetariano   

                                                  3  \
_id                                               4   
name                                           Luis   
surname                                       Gómez   
email                             user4@example.com   
password                

## Subida de CSV a BD Mongo

### 1. Churro a Mongo y consulta de Tablas en la BD de SAMOO

In [24]:
# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos
db = client["Samoo"]

# Obtener todas las colecciones en la base de datos
collections = db.list_collection_names()

# Contar y mostrar el número de documentos en cada colección
print("Número de documentos en cada colección de la base de datos 'Samoo':")
for collection_name in collections:
    collection = db[collection_name]
    num_documents = collection.count_documents({})
    print(f"Colección '{collection_name}': {num_documents} documentos")

# Cerrar la conexión
client.close()

Número de documentos en cada colección de la base de datos 'Samoo':
Colección 'events': 3 documentos
Colección 'meetings': 6 documentos
Colección 'supliers': 6 documentos
Colección 'collabs': 6 documentos
Colección 'users': 308 documentos
Colección 'places': 1 documentos


--> user aparece como 308 porque teniamos subidos 8 y le hemos añadido 300

### Borramos los datos de users y volvemos a cargos los 300 del csv

In [25]:
""" BASE DE DATOS USERS"""


# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos y la colección
db = client["Samoo"]
collection = db["users"]

# Supongamos que ya tienes un dataframe llamado df
# df = pd.read_csv("tu_archivo.csv")  # Ejemplo de cómo cargar un dataframe

# Convertir el dataframe a una lista de diccionarios
data = df_users.to_dict(orient="records")

# Eliminar todos los documentos existentes en la colección     OJOOOoooooOoOOOOOOOOOOOOOOOOOO , esta linead borra BD
collection.delete_many({})

# Insertar los datos en la colección
collection.insert_many(data)

print("Datos insertados exitosamente")

# Cerrar la conexión
client.close()

Datos insertados exitosamente


### comprobamos de nuevo la tabla users cuantas entradas tiene:

In [26]:
# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos
db = client["Samoo"]

# Obtener todas las colecciones en la base de datos
collections = db.list_collection_names()

# Contar y mostrar el número de documentos en cada colección
print("Número de documentos en cada colección de la base de datos 'Samoo':")
for collection_name in collections:
    collection = db[collection_name]
    num_documents = collection.count_documents({})
    print(f"Colección '{collection_name}': {num_documents} documentos")

# Cerrar la conexión
client.close()

Número de documentos en cada colección de la base de datos 'Samoo':
Colección 'events': 3 documentos
Colección 'meetings': 6 documentos
Colección 'supliers': 6 documentos
Colección 'collabs': 6 documentos
Colección 'users': 300 documentos
Colección 'places': 1 documentos


--> OK, ya hemos actualizado y contrastado como reemplazar los datos de una BD (users)

## 2. ACTUALIZACION DE TODAS LAS TABLAS CON LOS NUEVOS DATOS

In [27]:
"""BORRADO Y ACTUALIZACION DE LA TABLA PLACES"""


# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos y la colección
db = client["Samoo"]
collection = db["places"]

# Supongamos que ya tienes un dataframe llamado df
# df = pd.read_csv("tu_archivo.csv")  # Ejemplo de cómo cargar un dataframe

# Convertir el dataframe a una lista de diccionarios
data = df_places.to_dict(orient="records")

# Eliminar todos los documentos existentes en la colección     OJOOOoooooOoOOOOOOOOOOOOOOOOOO , esta linead borra BD
collection.delete_many({})

# Insertar los datos en la colección
collection.insert_many(data)

print("Datos insertados exitosamente")

# Cerrar la conexión
client.close()

Datos insertados exitosamente


In [30]:
"""BORRADO Y ACTUALIZACION DE LA TABLA EVENTS"""


# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos y la colección
db = client["Samoo"]
collection = db["events"]

# Supongamos que ya tienes un dataframe llamado df
# df = pd.read_csv("tu_archivo.csv")  # Ejemplo de cómo cargar un dataframe

# Convertir el dataframe a una lista de diccionarios
data = df_events.to_dict(orient="records")

# Eliminar todos los documentos existentes en la colección     OJOOOoooooOoOOOOOOOOOOOOOOOOOO , esta linead borra BD
collection.delete_many({})

# Insertar los datos en la colección
collection.insert_many(data)

print("Datos insertados exitosamente")

# Cerrar la conexión
client.close()


Datos insertados exitosamente


In [31]:
"""BORRADO Y ACTUALIZACION DE LA TABLA MEETINGS"""


# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos y la colección
db = client["Samoo"]
collection = db["meetings"]

# Supongamos que ya tienes un dataframe llamado df
# df = pd.read_csv("tu_archivo.csv")  # Ejemplo de cómo cargar un dataframe

# Convertir el dataframe a una lista de diccionarios
data = df_meetings.to_dict(orient="records")

# Eliminar todos los documentos existentes en la colección     OJOOOoooooOoOOOOOOOOOOOOOOOOOO , esta linead borra BD
collection.delete_many({})

# Insertar los datos en la colección
collection.insert_many(data)

print("Datos insertados exitosamente")

# Cerrar la conexión
client.close()


Datos insertados exitosamente


In [32]:
"""BORRADO Y ACTUALIZACION DE LA TABLA suppliers"""


# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos y la colección
db = client["Samoo"]
collection = db["suppliers"]

# Supongamos que ya tienes un dataframe llamado df
# df = pd.read_csv("tu_archivo.csv")  # Ejemplo de cómo cargar un dataframe

# Convertir el dataframe a una lista de diccionarios
data = df_suppliers.to_dict(orient="records")

# Eliminar todos los documentos existentes en la colección     OJOOOoooooOoOOOOOOOOOOOOOOOOOO , esta linead borra BD
collection.delete_many({})

# Insertar los datos en la colección
collection.insert_many(data)

print("Datos insertados exitosamente")

# Cerrar la conexión
client.close()


Datos insertados exitosamente


YA TENEMOS TODAS LAS TABLAS ACTUALIZADAS EN LA BD

COMPROBAMOS DE NUEVO EL RECUENTO DE MUESTRAS EN CADA TABLA: 

In [33]:
# URL del repositorio de MongoDB
mongo_url = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"

# Conectar al cliente de MongoDB
client = MongoClient(mongo_url)

# Seleccionar la base de datos
db = client["Samoo"]

# Obtener todas las colecciones en la base de datos
collections = db.list_collection_names()

# Contar y mostrar el número de documentos en cada colección
print("Número de documentos en cada colección de la base de datos 'Samoo':")
for collection_name in collections:
    collection = db[collection_name]
    num_documents = collection.count_documents({})
    print(f"Colección '{collection_name}': {num_documents} documentos")

# Cerrar la conexión
client.close()

Número de documentos en cada colección de la base de datos 'Samoo':
Colección 'events': 25 documentos
Colección 'meetings': 540 documentos
Colección 'supliers': 6 documentos
Colección 'collabs': 6 documentos
Colección 'suppliers': 25 documentos
Colección 'users': 300 documentos
Colección 'places': 2 documentos


-> SUPLIERS Y COLLABS DEBEMOS DE BORRARLAS, LO HAREMOS DIRECTAMENTE DESDE LA BBDD, NO SIN ANTES CONSULTAR CON FULL  STACK SI LES PUEDE AFECTAR

# A PARTIR DE AQUI CODIGO ANTIGUO

In [ ]:
db = client["Samoo"]
collection = db["collabs"]
dat = list(collection.find())
dff_collabs = pd.DataFrame(dat)
dff_collabs

In [24]:
""" TABLA USERS"""

from pymongo import MongoClient
# import pandas as pd

# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "users"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_users = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el DataFrame
dff_users

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


_id     name                         email  \
0   667407aa4e3870cf46cb5d26     Hola                hola@gmail.com   
1   667410d6412a0a10b853a236  pedrito                pedrito@rok.es   
2   66741880412a0a10b853a239   Manuel  manueldanapallardo@gmail.com   
3   66741949412a0a10b853a23c    Pedro          pedro@picapiedra.rok   
4   66741ae3378ab4065a7bfd7c      NaN                           NaN   
5   66741ae3378ab4065a7bfd7d      NaN                           NaN   
6   66741ae3378ab4065a7bfd7e      NaN                           NaN   
7   66741ae3378ab4065a7bfd7f      NaN                           NaN   
8   66741ae3378ab4065a7bfd80      NaN                           NaN   
9   66742c6f9c588465fddbe57e     pepe                pepe@gmail.com   
10  66742e58378ab4065a7bfd82      NaN                           NaN   
11  66742e58378ab4065a7bfd83      NaN                           NaN   
12  66742e58378ab4065a7bfd84      NaN                           NaN   
13  66742e58378ab4065a7bfd85      NaN                           NaN   
14  66742e58378ab4065a7bfd86      NaN                           NaN   

                                             password  \
0   $2b$10$y.9eXytVaFJqNwgcqvnpM.UuTPIsWpOQNl0FJVh...   
1   $2b$10$CPak6SO3HNSV1g50m1vvGeCm/bhD/lKUe.zgI4Y...   
2   $2b$10$6CKx9wAYt7LSlP2fJeR0nuJXgFDsbXdKfXUocw2...   
3   $2b$10$nJHdDCtcrE6rPWRHfjvpee8sS1ovzSAbc423jSe...   
4                                                 NaN   
5                                                 NaN   
6                                                 NaN   
7                                                 NaN   
8                                                 NaN   
9   $2b$10$y3xH3hIxSAoONel2WHybTu17QGNevvjPxbLOxcA...   
10                                                NaN   
11                                                NaN   
12                                                NaN   
13                                                NaN   
14                                                NaN   

                                               tokens               createdAt  \
0   [eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQi... 2024-06-20 10:42:50.952   
1                                                  [] 2024-06-20 11:21:58.716   
2   [eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQi... 2024-06-20 11:54:40.774   
3                                                  [] 2024-06-20 11:58:01.323   
4                                                 NaN                     NaT   
5                                                 NaN                     NaT   
6                                                 NaN                     NaT   
7                                                 NaN                     NaT   
8                                                 NaN                     NaT   
9                                                  [] 2024-06-20 13:19:43.857   
10                                                NaN                     NaT   
11                                                NaN                     NaT   
12                                                NaN                     NaT   
13                                                NaN                     NaT   
14                                                NaN                     NaT   

                 updatedAt  __v confirmed  id_user  ... prefijo_telefono  \
0  2024-06-20 10:45:49.490  1.0      True      NaN  ...              NaN   
1  2024-06-20 11:21:58.716  0.0       NaN      NaN  ...              NaN   
2  2024-06-20 13:37:47.958  2.0       NaN      NaN  ...              NaN   
3  2024-06-20 11:58:01.323  0.0       NaN      NaN  ...              NaN   
4                      NaT  NaN       NaN      1.0  ...             34.0   
5                      NaT  NaN       NaN      2.0  ...             34.0   
6                      NaT  NaN       NaN      3.0  ...             34.0   
7                      NaT  NaN       NaN      4.0  ...             34

In [25]:
dff_users.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 23 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   _id                 15 non-null     object        
 1   name                5 non-null      object        
 2   email               5 non-null      object        
 3   password            5 non-null      object        
 4   tokens              5 non-null      object        
 5   createdAt           5 non-null      datetime64[ns]
 6   updatedAt           5 non-null      datetime64[ns]
 7   __v                 5 non-null      float64       
 8   confirmed           1 non-null      object        
 9   id_user             10 non-null     float64       
 10  nombre              10 non-null     object        
 11  apellidos           10 non-null     object        
 12  correo_electronico  10 non-null     object        
 13  prefijo_telefono    10 non-null     float64       
 

In [13]:
""" TABLA COLLABS"""

from pymongo import MongoClient
# import pandas as pd

# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "collabs"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_collabs = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el Dat.inaFrame
dff_collabs.info()

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   _id                3 non-null      object
 1   id_collab          3 non-null      int64 
 2   nombre_empresa     3 non-null      object
 3   direccion_empresa  3 non-null      object
 4   correo_empresa     3 non-null      object
 5   telefono_empresa   3 non-null      int64 
 6   tipo_collab        3 non-null      object
 7   id_users           3 non-null      object
dtypes: int64(2), object(6)
memory usage: 320.0+ bytes


In [14]:
""" TABLA EVENTOS"""
from pymongo import MongoClient
# import pandas as pd

# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "events"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_events = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el Dat.inaFrame
dff_events.info()

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 5 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   _id                  5 non-null      object
 1   id_user              5 non-null      int64 
 2   alergenos            5 non-null      object
 3   preferencias_comida  5 non-null      object
 4   intereses            5 non-null      object
dtypes: int64(1), object(4)
memory usage: 328.0+ bytes


In [15]:
""" TABLA MEETINGS"""


# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "meetings"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_meetings = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el Dat.inaFrame
dff_meetings.info()

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   _id         3 non-null      object
 1   id_meeting  3 non-null      int64 
 2   id_collab   3 non-null      int64 
 3   id_user     3 non-null      int64 
 4   fecha       3 non-null      object
 5   hora        3 non-null      object
dtypes: int64(3), object(3)
memory usage: 272.0+ bytes


In [16]:
""" TABLA SUPLIERS"""


# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "supliers"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_supliers = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el Dat.inaFrame
dff_supliers.info()

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   _id        2 non-null      object        
 1   cif        2 non-null      object        
 2   name       2 non-null      object        
 3   password   2 non-null      object        
 4   employes   2 non-null      int64         
 5   tokens     2 non-null      object        
 6   createdAt  2 non-null      datetime64[ns]
 7   updatedAt  2 non-null      datetime64[ns]
 8   __v        2 non-null      int64         
dtypes: datetime64[ns](2), int64(2), object(5)
memory usage: 272.0+ bytes


In [17]:
""" TABLA PLACES"""


# Definir la función para descargar la colección desde MongoDB
def download_collection_from_mongodb(uri, db_name, collection_name):
    # Conectar a MongoDB
    client = MongoClient(uri)
    try:
        # Seleccionar la base de datos y la colección
        db = client[db_name]
        collection = db[collection_name]

        # Descargar los datos de la colección
        data = list(collection.find())

        # Convertir los datos a un DataFrame de pandas
        df = pd.DataFrame(data)
    finally:
        # Cerrar la conexión
        client.close()
    
    return df

# Datos de conexión a MongoDB
uri = "mongodb+srv://Cluster78620:uuMda3E0UnvIPzdi@cluster786∫20.jcfu9lt.mongodb.net"  # Cambia la URL según sea necesario
db_name = "Samoo"  # Cambia 'nombre_de_la_base_de_datos' por el nombre de tu base de datos
collection_name = "places"  # Cambia 'nombre_de_la_coleccion' por el nombre de tu colección

# Descargar la colección y convertirla en un DataFrame
dff_places = download_collection_from_mongodb(uri, db_name, collection_name)

# Mostrar el Dat.inaFrame
dff_places.info()

# Opcional: Guardar el DataFrame en un archivo CSV
# df.to_csv('nombre_del_archivo.csv', index=False)  # Cambia 'nombre_del_archivo.csv' por el nombre deseado


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   _id          4 non-null      object
 1   id_place     4 non-null      int64 
 2   nombre_sala  4 non-null      object
 3   capacidad    4 non-null      int64 
dtypes: int64(2), object(2)
memory usage: 256.0+ bytes
